doc/source/groupby.rst

```diff
@@ -160,6 +160,47 @@ only verifies that you've passed a valid mapping.
    GroupBy operations (though can't be guaranteed to be the most
    efficient). You can get quite creative with the label mapping functions.
 
+.. _groupby.sorting:
+
+GroupBy sorting
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default the group keys are sorted during the ``groupby`` operation. You may however pass ``sort=False`` for potential speedups:
+
+.. ipython:: python
+
+   df2 = pd.DataFrame({'X' : ['B', 'B', 'A', 'A'], 'Y' : [1, 2, 3, 4]})
+   df2.groupby(['X']).sum()
+   df2.groupby(['X'], sort=False).sum()
+
+
+Note that ``groupby`` will preserve the order in which *observations* are sorted *within* each group. For example, the groups created by ``groupby()`` below are in the order the appeared in the original ``DataFrame``:
+
+.. ipython:: python
+
+   df3 = pd.DataFrame({'X' : ['A', 'B', 'A', 'B'], 'Y' : [1, 4, 3, 2]})
+   df3.groupby(['X']).get_group('A')
+
+   df3.groupby(['X']).get_group('B')
+
+
+.. note::
+
+   Users should be careful about re-sorting target objects after execution of ``groupby()``, as these changes may not be reflected within the ``groupby()`` object. For example:
+
+   .. ipython:: python
+
+      g = df3.groupby(['X'])
+
+      df3 = df3.sort_values('Y')
+      df3
+
+      g.get_group('A')
+
+      g.get_group('B')
+
```

Comment:
hmm, this is True, but I think the average user will find this a bit confusing. Can you expand/reword a bit. (its an important point), but not sure how to describe it.